frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -323,135 +370,191 @@ export const OperandList: React.FC<OperandListProps> = (props) => {
+  }>(watchedResources);
 
   // Refresh API definitions if at least one API is missing a model and definitions have not already been refreshed.
-  const apiMightBeOutdated = !inFlight && firehoseResources.length < providedAPIs.length;
+  const apiMightBeOutdated =
+    !inFlight && Object.keys(watchedResources).length < providedAPIs.length;
   React.useEffect(() => {
     if (!apiRefreshed && apiMightBeOutdated) {
       dispatch(getResources());
       setAPIRefreshed(true);
     }
   }, [apiMightBeOutdated, apiRefreshed, dispatch]);
 
-  if (inFlight) {
-    return null;
-  }
-
-  const EmptyMsg = () => (
-    <MsgBox
-      title={t('olm~No provided APIs defined')}
-      detail={t('olm~This application was not properly installed or configured.')}
-    />
-  );
-  const createLink = (name: string) =>
-    `/k8s/ns/${obj.metadata.namespace}/${ClusterServiceVersionModel.plural}/${
-      obj.metadata.name
-    }/${referenceForProvidedAPI(
-      _.find(providedAPIs, {
-        name,
-      }),
-    )}/~new`;
-  const createProps =
+  const createItems =
     providedAPIs.length > 1
-      ? {
-          items: providedAPIs.reduce(
-            (acc, api) => ({
-              ...acc,
-              [api.name]: api.displayName,
-            }),
-            {},
-          ),
-          createLink,
-        }
-      : {
-          to: providedAPIs.length === 1 ? createLink(providedAPIs[0].name) : null,
-        };
-
-  const owners = (ownerRefs: OwnerReference[], items: K8sResourceKind[]) =>
-    ownerRefs.filter(({ uid }) => items.filter(({ metadata }) => metadata.uid === uid).length > 0);
-  const flatten: Flatten<{
-    [key: string]: K8sResourceCommon[];
-  }> = (resources) =>
-    _.flatMap(resources, (resource) => _.map(resource.data, (item) => item)).filter(
-      ({ kind, metadata }, i, allResources) =>
-        providedAPIs.filter((item) => item.kind === kind).length > 0 ||
-        owners(metadata.ownerReferences || [], allResources).length > 0,
+      ? providedAPIs.reduce((acc, api) => ({ ...acc, [api.name]: api.displayName }), {})
+      : null;
+
+  const createNavigate = (name) =>
+    history.push(
+      `/k8s/ns/${obj.metadata.namespace}/${ClusterServiceVersionModel.plural}/${
+        obj.metadata.name
+      }/${referenceForProvidedAPI(_.find(providedAPIs, { name }))}/~new`,
     );
 
+  const data = React.useMemo(() => flatten(resources), [resources, flatten]);
+
   const rowFilters =
-    firehoseResources.length > 1
+    Object.keys(watchedResources).length > 1
       ? [
           {
-            filterGroupName: 'Resource Kind',
+            filterGroupName: t('Resource Kind'),
             type: 'clusterserviceversion-resource-kind',
             reducer: ({ kind }) => kind,
-            items: firehoseResources.map(({ kind }) => ({
+            items: Object.keys(watchedResources).map((kind) => ({
               id: kindForReference(kind),
               title: kindForReference(kind),
             })),
```

Comment:
the filter function
```suggestion
            items: Object.keys(watchedResources).map((kind) => ({
              id: kindForReference(kind),
              title: kindForReference(kind),
            })),
            filter: (filters, resource) => {
              if (!filters || !filters.selected || !filters.selected.length) {
                return true;
              }
              return filters.selected.includes(resource.kind);
            },
```

but please try it, I did not :)